RecoTracker/GeometryESProducer/plugins/TrackerMTDRecoGeometryESProducer.cc

```diff
@@ -0,0 +1,76 @@
+#include "Geometry/TrackerNumberingBuilder/interface/GeometricDet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+#include "RecoTracker/TkDetLayers/interface/GeometricSearchTrackerBuilder.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeometry.h"
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+#include "Geometry/Records/interface/MTDTopologyRcd.h"
+#include "Geometry/Records/interface/MTDDigiGeometryRecord.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include <memory>
+#include <string>
+
+class TrackerMTDRecoGeometryESProducer : public edm::ESProducer {
+public:
+  TrackerMTDRecoGeometryESProducer(const edm::ParameterSet &p);
+
+  std::unique_ptr<GeometricSearchTracker> produce(const TrackerRecoGeometryRecord &);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopToken_;
+  edm::ESGetToken<MTDGeometry, MTDDigiGeometryRecord> mtdgeomToken_;
+  edm::ESGetToken<MTDTopology, MTDTopologyRcd> mtdTopToken_;
+  bool usePhase2Stacks_;
+};
+
+using namespace edm;
+
+TrackerMTDRecoGeometryESProducer::TrackerMTDRecoGeometryESProducer(const edm::ParameterSet &p)
+    : usePhase2Stacks_(p.getParameter<bool>("usePhase2Stacks")) {
+  auto c = setWhatProduced(this);
+
+  // 08-Oct-2007 - Patrick Janot
```

Comment:
I understand that this was originally derived from RecoTracker/GeometryESProducer/plugins/TrackerRecoGeometryESProducer.cc
However, the comments copy-pasted here do not fully apply to the current case: either update them, or simply remove them, in order to avoid possible future misunderstandings

Reply:
Hi Andrea,

Thanks. Yes, I have removed them in a later commit.